### PR TITLE
Add support for App Mesh Preview Environment

### PR DESCRIPTION
**Pull request by stefanprodan**
This PR adds support for App Mesh preview environment. Fix: #2299

Changes:
- add `appMeshPreview` addon policy
- add `appmesh-preview-access` flag to node groups

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [x] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

**Comment by stefanprodan** [comment]
running `make check-all-generated-files-up-to-date` locally look ok, no idea what's going on @martina-if

**Comment by martina-if** [comment]
@stefanprodan I think it could be because we've been upgrading different tools. Can you run `make install-build-deps` and `make generate-all` again? (I can do it for you and commit the changes if you want but it would be good if you have it correctly set up as well)

**Comment by stefanprodan** [comment]
@martina-if thanks! I had the old tooling installed, should be ok now.

File: pkg/cfn/builder/api_test.go [hunk 1]
```diff
@@ -1136,6 +1137,49 @@ var _ = Describe("CloudFormation template builder API", func() {
...
+				"servicediscovery:ListNamespaces",
+				"servicediscovery:ListServices",
+				"servicediscovery:GetInstancesHealthStatus",
+				"servicediscovery:UpdateInstanceCustomHealthStatus",
+				"servicediscovery:GetOperation",
+				"route53:GetHealthCheck",
+				"route53:CreateHealthCheck",
+				"route53:UpdateHealthCheck",
+				"route53:ChangeResourceRecordSets",
+				"route53:DeleteHealthCheck",
```
Comment (martina-if) [inline pkg/cfn/builder/api_test.go:1177]: Can we extract these into a common array for both tests so it's easier to see in a quick look what is the difference between appmesh and appmesh-preview?

File: pkg/cfn/builder/iam_helper.go [hunk 2]
```diff
@@ -90,28 +90,33 @@ func createRole(cfnTemplate cfnTemplate, iamConfig *api.NodeGroupIAM, managed bo
...
-				"servicediscovery:GetOperation",
-				"route53:GetHealthCheck",
-				"route53:CreateHealthCheck",
-				"route53:UpdateHealthCheck",
-				"route53:ChangeResourceRecordSets",
-				"route53:DeleteHealthCheck",
-			},
+			append(appMeshActions, "appmesh:*"),
+		)
+	} else if api.IsEnabled(iamConfig.WithAddonPolicies.AppMeshPreview) {
```
Comment (martina-if) [inline pkg/cfn/builder/iam_helper.go:117]: I think if both flags are allowed at the same time both actions should be included. If only one of them is going to be included in the policy the UX should fail when both flags or fields are provided (as they are incompatible).

Comment (stefanprodan) [reply pkg/cfn/builder/iam_helper.go]: I've reconsidered this, both flags could be specified as one may want to switch between stable and preview on the same cluster.

File: pkg/cfn/builder/api_test.go (hunk 1 above)
Comment (stefanprodan) [reply pkg/cfn/builder/api_test.go]: Done!

**Comment by stefanprodan** [comment]
@sayboras I've updated the yaml, thanks.

**Comment by bcelenza** [comment]
Thanks for tackling this @stefanprodan !

**Pull request merged**